## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 436852`
Commit: `de54227ba194d3d06c54d897d6d2f0b57df0d05e`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.cachier</li>
    <li>python312Packages.cachier.dist</li>
    <li>python313Packages.cachier</li>
    <li>python313Packages.cachier.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.cachier</li>
    <li>python312Packages.cachier.dist</li>
    <li>python313Packages.cachier</li>
    <li>python313Packages.cachier.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.cachier</li>
    <li>python312Packages.cachier.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.cachier</li>
    <li>python313Packages.cachier.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.cachier</li>
    <li>python312Packages.cachier.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python313Packages.cachier</li>
    <li>python313Packages.cachier.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.cachier</summary>
<pre>ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/hqrh8g2nz70m82pmclpp99w96njlvni2-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-0/tmp.GhgH4R9ngB/.cachier/', is_recursive=True> - it is already scheduled
ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/hqrh8g2nz70m82pmclpp99w96njlvni2-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-0/tmp.GhgH4R9ngB/.cachier/', is_recursive=True> - it is already scheduled
ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/hqrh8g2nz70m82pmclpp99w96njlvni2-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-0/tmp.GhgH4R9ngB/.cachier/', is_recursive=True> - it is already scheduled
=========================== short test summary info ============================
SKIPPED [1] tests/test_redis_core.py:185: Redis not available
FAILED tests/test_pickle_core.py::test_bad_cache_file[False] - assert False is not False
FAILED tests/test_pickle_core.py::test_delete_cache_file[False] - assert False is not False
====== 2 failed, 91 passed, 1 skipped, 37 deselected in 317.66s (0:05:17) ======</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.cachier</summary>
<pre>ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/pnn44mz2p86vql7909jkbkykfznwyw4r-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-1/tmp.acf0KM2jGs/.cachier/', is_recursive=True> - it is already scheduled
ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/pnn44mz2p86vql7909jkbkykfznwyw4r-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-1/tmp.acf0KM2jGs/.cachier/', is_recursive=True> - it is already scheduled
ERROR    fsevents:fsevents.py:310 Unhandled exception in FSEventsEmitter
Traceback (most recent call last):
  File "/nix/store/pnn44mz2p86vql7909jkbkykfznwyw4r-python3.12-watchdog-6.0.0/lib/python3.12/site-packages/watchdog/observers/fsevents.py", line 307, in run
    _fsevents.add_watch(self, self.watch, self.events_callback, self.pathnames)
RuntimeError: Cannot add watch <ObservedWatch: path='/private/tmp/nix-build-python3.12-cachier-4.1.0.drv-1/tmp.acf0KM2jGs/.cachier/', is_recursive=True> - it is already scheduled
=========================== short test summary info ============================
SKIPPED [1] tests/test_redis_core.py:185: Redis not available
FAILED tests/test_pickle_core.py::test_bad_cache_file[False] - assert False is not False
FAILED tests/test_pickle_core.py::test_delete_cache_file[False] - assert False is not False
====== 2 failed, 91 passed, 1 skipped, 37 deselected in 317.31s (0:05:17) ======</pre>
</details>
</details>

--- a/pkgs/development/python-modules/cachier/default.nix
+++ b/pkgs/development/python-modules/cachier/default.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   pythonOlder,
   fetchFromGitHub,
@@ -9,6 +10,7 @@
   portalocker,
   pytestCheckHook,
   pytest-cov-stub,
+  sqlalchemy,
   pymongo,
   dnspython,
   pymongo-inmemory,
@@ -46,6 +48,7 @@ buildPythonPackage rec {
   nativeCheckInputs = [
     pytestCheckHook
     pytest-cov-stub
+    sqlalchemy
     pymongo
     dnspython
     pymongo-inmemory
@@ -67,9 +70,23 @@ buildPythonPackage rec {
     # don't test formatting
     "test_flake8"
 
+    # slow, spawns 800+ threads
+    "test_inotify_instance_limit_reached"
+
     # timing sensitive
     "test_being_calc_next_time"
     "test_pickle_being_calculated"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # sensitive to host file system
+    # Unhandled exception in FSEventsEmitter -  RuntimeError: Cannot add watch - it is already scheduled
+    "test_bad_cache_file"
+    "test_delete_cache_file"
+  ];
+
+  disabledTestPaths = [
+    # Keeps breaking due to concurrent access or failing to close the db between tests.
+    "tests/test_sql_core.py"
   ];
 
   preBuild = ''

